### PR TITLE
[MISC] deprecate gap_scheme

### PR DESCRIPTION
**Pull request by SGSSGene**
This adds [DEPRECATE] and \deprecate annotation to gap_scheme or related
classes.
Also removed gap policy "make_cache" since it didn't exist and is using
gap_scheme.

### CODE REVIEW

**Comment by SGSSGene** [comment]
Open: can the header file be declared "deprecated"? And if yes add `SEQAN3_DEPRECATED_HEADER`

**Comment by SGSSGene** [comment]
> Open: can the header file be declared "deprecated"? And if yes add `SEQAN3_DEPRECATED_HEADER`

I don't think the header file itself is deprecated. There is no new header file to be used.

**Review by Irallia — APPROVED** [review]
LGFM, thank you!

File: include/seqan3/alignment/scoring/gap_scheme.hpp [hunk 1]
```diff
@@ -26,8 +26,8 @@ namespace seqan3
 // seqan3::gap_score
 // ------------------------------------------------------------------
 
-/*!\brief A strong type of underlying type `score_type` that represents the score of any character against a gap
- *        character. 
+/*!\brief [DEPRECATED] A strong type of underlying type `score_type` that
+ *        represents the score of any character against a gap character.
  * \deprecated This type is deprecated and will be removed in SeqAn3.1.
  * \tparam score_type The underlying type.
  * \ingroup scoring
```
Comment (marehr) [inline include/seqan3/alignment/scoring/gap_scheme.hpp:33]: You marked this as deprecated, but `struct SEQAN3_DEPRECATED_310 gap_score` isn't deprecated, is there a reason for that?

File: include/seqan3/alignment/scoring/gap_scheme.hpp [hunk 2]
```diff
@@ -53,8 +53,9 @@ gap_score(score_type) -> gap_score<score_type>;
...
 // ------------------------------------------------------------------
 
-/*!\brief A strong type of underlying type `score_type` that represents an additional score (usually negative) that
- *        is incurred once additionaly per stretch of consecutive gaps.
+/*!\brief [DEPRECATED] A strong type of underlying type `score_type` that
+ *        represents an additional score (usually negative) that is incurred
+ *        once additionaly per stretch of consecutive gaps.
  * \deprecated This type is deprecated and will be removed in SeqAn3.1.
  * \tparam score_type The underlying type.
  * \ingroup scoring
```
Comment (marehr) [inline include/seqan3/alignment/scoring/gap_scheme.hpp:61]: Here too?

File: include/seqan3/alignment/scoring/gap_scheme.hpp (hunk 1 above)
Comment (SGSSGene) [reply include/seqan3/alignment/scoring/gap_scheme.hpp:33]: Ah, nice, I missed that. :-).  git log shows that gap_score was marked deprecated already last year see commit 54a2ea6a

**Review by marehr — APPROVED** [review]

**Pull request merged**